modules/api/src/main/scala/com/github/scoquelin/arugula/commands/RedisHashAsyncCommands.scala

```diff
@@ -1,19 +1,165 @@
 package com.github.scoquelin.arugula.commands
 
+import scala.collection.immutable.ListMap
 import scala.concurrent.Future
 
+import com.github.scoquelin.arugula.commands.RedisKeyAsyncCommands.ScanCursor
+
 /**
  * Asynchronous commands for manipulating/querying Hashes (key/value pairs)
  *
  * @tparam K The key type
  * @tparam V The value type
  */
 trait RedisHashAsyncCommands[K, V] {
+  /**
+   * Delete one or more hash fields
+   * @param key The key
+   * @param fields The fields to delete
+   * @return The number of fields that were removed from the hash
+   */
   def hDel(key: K, fields: K*): Future[Long]
+
+  /**
+   * Determine if a hash field exists
+   * @param key The key
+   * @param field The field
+   * @return True if the field exists, false otherwise
+   */
+  def hExists(key: K, field: K): Future[Boolean]
+
+  /**
+   * Get the value of a hash field
+   * @param key The key
+   * @param field The field
+   * @return The value of the field, or None if the field does not exist
+   */
   def hGet(key: K, field: K): Future[Option[V]]
+
+  /**
+   * get all the keys in a hash
+   * @param key The key
+   * @return A list of all the keys in the hash
+   */
+  def hKeys(key: K): Future[Seq[K]]
```

Comment:
TLDR : I would recommend having a more fine grained type in the interface like `List`, because `Seq` is simply too generic and can lead to weird corner cases.

Explanation: Here we are controlling the implementation but let's say we choose to implement it as a `Stream` (which will conform to `Seq`), if the caller (who does not know the underlying implementation details) of the API calls `toList`, the call might never terminate when dealing with infinite collections...